python/feldera/pipeline_builder.py

```diff
@@ -0,0 +1,155 @@
+from typing import Self
+
+from feldera.rest.feldera_client import FelderaClient
+from feldera.rest.pipeline import Pipeline as InnerPipeline
+from feldera.pipeline import Pipeline
+from feldera.enums import CompilationProfile
+from feldera.runtime_config import RuntimeConfig, Resources
+from feldera.rest.errors import FelderaAPIError
+
+
+class PipelineBuilder:
+    """
+    A builder for creating a Pipeline in Feldera.
+
+    :param client: The `.FelderaClient` instance
+    :param name: The name of the pipeline
+    :param description: The description of the pipeline
+    :param sql: The SQL code of the pipeline
+    :param compilation_profile: The compilation profile to use
+    :param runtime_config: The runtime config to use
+    """
+
+    def __init__(
```

Comment:
Later validation at creation time with this change becomes unnecessary.

Reply:
I liked the idea of not requiring them to be available at the same time. 

Such that you can do things like:


```py

builder = PipelineBuilder(client).with_name(name)

# do other stuff

# dynamically get the connector info and build the SQL code
sql = generate_sql(...)

pipeline = builder.with_sql(sql).create()

```

Reply:
But why would `builder` be needed in the `#do other stuff` part?

Reply:
```
# do other stuff

# dynamically get the connector info and build the SQL code
sql = generate_sql(...)

builder = PipelineBuilder(client, name, sql)
```

Reply:
It isn't necessary there. 
I am a little colored with our older API creating the SQLContext first such that all the setup code is dealt with first, which felt kinda nice.

Reply:
I agree it made sense before with `SQLContext` as first having to build up the pipeline with a program, connectors, attaching them etc., but now with the SQL having everything in it, it seems unnecessary as there is almost no setup code or logic needed. @ryzhyk what are your thoughts?

Reply:
It's probably ok to get rid of all the `with_xxx` methods and just make them arguments to the constructor. With separate methods, we're simulating the Rust builder pattern, which avoids having constructors with many arguments. In Python this isn't really an issue because we can use named arguments instead. So I guess I'm ok with just a single constructor with multiple arguments.